addons/ai/fnc_taskPatrol.sqf

```diff
@@ -46,25 +51,30 @@ _position = _position call CBA_fnc_getPos;
 // Clear existing waypoints first
 [_group] call CBA_fnc_clearWaypoints;
 
-private _this =+ _this;
-switch (count _this) do {
-    case 1 : {_this append [_position, _radius]};
-    case 2 : {_this pushBack _radius};
-    default {};
-};
+// Can pass paremeters straight through to addWaypoint
+_this =+ _this;
+_this set [2,0];
 if (count _this > 3) then {
     _this deleteAt 3;
 };
 
-// Store first WP to close loop later
-private _wp = _this call CBA_fnc_addWaypoint;
+// Using angles create better patrol patterns
+// Also fixes weird editor bug where all WP are on same position
+private _step = 360/_count;
+private _offset = random _step;
+for "_i" from 1 to _count do {
+    // Gaussian distribution avoids all waypoints ending up in the center
+    private _rad = _radius*random [0.1, 0.75, 1];
+
+    // Alternate sides of circle & modulate offset
+    private _theta = (_i%2)*180 + sin(deg(_step*_i))*_offset + _step*_i;
```

Comment:
No specific reason, this is just how I tend to write calculations (probably because I'm used to seeing algebraic expressions all the time in engineering)